experiments/gromacs/openmp/ramble.yaml

```diff
@@ -0,0 +1,50 @@
+ramble:
+  include:
+  - ./configs/spack.yaml
+  - ./configs/variables.yaml
+
+  config:
+    deprecated: true
+    spack_flags:
+      install: '--add --keep-stage'
+      concretize: '-U -f'
+
+  applications:
+    gromacs:
+      workloads:
+        water_gmx50:
+          env_vars:
+            set:
+              OMP_NUM_THREADS: '{omp_num_threads}'
+          variables:
+            n_ranks: '{processes_per_node} * {n_nodes}'
+            size: ['1536', '3072']
+            experiment_setup: ''
+            processes_per_node: ['8', '4']
+            n_nodes: ['1', '2']
+            threads_per_node_core: ['2', '4']
+            omp_num_threads: '{threads_per_node_core} * {n_nodes}'
```

Comment:
Can you explain what this means? Isn't `omp_num_threads` here the same as `OMP_NUM_THREADS`, i.e. the threads per MPI rank?

Reply:
OMP_NUM_THREADS (set using the value of omp_num_threads) is the threads per MPI rank. For this particular experiment config we scale omp_num_threads with n_nodes - for 1 node we use 2 and 4 threads, while for 2 nodes we use 4 and 8 threads. In both cases the total number of MPI ranks is 8, but since the latter case uses 2 nodes, we can assign the additional cores to more omp threads. This can be changed per the experiment's requirements

Reply:
OK, I see. That's a atypical way of scaling to multiple nodes in MD/GROMACS. I suggest to using a fixed `OMP_NUM_THREADS`  and scale the number of ranks instead. On CPU-only systems typically small values of  `OMP_NUM_THREADS`  are best (1-3) until the case starts running out of parallelism (limiting domain decomposition) and increasing the  `OMP_NUM_THREADS`  can help to keep the number of ranks fixed while still using more resources.

On GPU-accelerated machines, the decomposition, hence the rank count is determined by the number of GPU devices (typically 1 or 2 ranks per device).